tutorials/Land/Soil/PhaseChange/phase_change_analytic_test.jl

```diff
@@ -0,0 +1,437 @@
+# for further information on the solution. It takes the form of a function
+# for T(z) on each side of the freezing front interface, which depends on
+# the thermal properties in that region, and which is also parameterized by
+# a parameter (ζ), which we show how to solve for below. In computing the
+# thermal properties, we evaluate the conductivity and heat capacity
+# assuming that all of the water is either in liquid or frozen form, with total
+# mass proportional to ``θ_{l,0}ρ_l`` (as we have no water diffusion).
+
+# Compute the thermal conductivity and heat capacity in the frozen region - subscript 1.
+θ_l0 = FT(0.33)
+kdry = k_dry(param_set, soil_param_functions)
+ksat = saturated_thermal_conductivity(
+    FT(0.0),
+    θ_l0 * ρ_cloud_liq(param_set) / ρ_cloud_ice(param_set),
+    κ_sat_unfrozen,
+    κ_sat_frozen,
+)
+kersten = kersten_number(
+    θ_l0 * ρ_cloud_liq(param_set) / ρ_cloud_ice(param_set),
+    θ_l0 * ρ_cloud_liq(param_set) / ρ_cloud_ice(param_set) / porosity,
+    soil_param_functions,
+)
+λ1 = thermal_conductivity(kdry, kersten, ksat)
+c1 = volumetric_heat_capacity(
+    FT(0.0),
+    θ_l0 * ρ_cloud_liq(param_set) / ρ_cloud_ice(param_set),
+    ρc_ds,
+    param_set,
+)
+d1 = λ1 / c1;
+
+# Compute the thermal conductivity and heat capacity in the region
+# with liquid water - subscript 2.
+ksat =
+    saturated_thermal_conductivity(θ_l0, FT(0.0), κ_sat_unfrozen, κ_sat_frozen)
+kersten = kersten_number(FT(0.0), θ_l0 / porosity, soil_param_functions)
+λ2 = thermal_conductivity(kdry, kersten, ksat)
+c2 = volumetric_heat_capacity(θ_l0, FT(0.0), ρc_ds, param_set)
+d2 = λ2 / c2;
+
+# Initial T and surface T, in Celsius
+Ti = FT(2)
+Ts = FT(-10.0);
+
+# The solution requires the root of the implicit equation below
+# (you'll need to install `Roots` via the package manager).
+# ``` julia
+# using Roots
+# function implicit(ζ)
+#    term1 = exp(-ζ^2) / ζ / erf(ζ)
+#    term2 =
+#        -λ2 * sqrt(d1) * (Ti - 0) /
+#        (λ1 * sqrt(d2) * (0 - Ts) * ζ * erfc(ζ * sqrt(d1 / d2))) *
+#        exp(-d1 / d2 * ζ^2)
+#    term3 =
+#        -LH_f0(param_set) * ρ_cloud_liq(param_set) * θ_l0 * sqrt(π) / c1 /
+#        (0 - Ts)
+#    return (term1 + term2 + term3)
+# end
+# find_zero(implicit, (0.25,0.27), Bisection())
+# ```
+
+# The root is
+ζ = 0.26447353269809687;
+
+
+
+# This function plots the analytic solution
+# and the simulated result, at the output time index
+# `k`.
+function f(k; ζ = ζ)
+    T = all_data[k]["soil.heat.T"][:] .- 273.15
+    plot(
+        T,
+        z[:],
+        xlim = [-10.5, 3],
+        ylim = [zmin, zmax],
+        xlabel = "T",
+        label = "simulation",
+    )
+    t = time_data[k]
+    zf = 2.0 * ζ * sqrt(d1 * t)
+    myz = zmin:0.01:0
+    spatially_varying =
+        (erfc.(abs.(myz) ./ (zf / ζ / (d1 / d2)^0.5))) ./
+        erfc(ζ * (d1 / d2)^0.5)
+    mask = abs.(myz) .> zf
```

Comment:
Looking at the plot output in the rendered tutorials, It looks like one of these masking conditions has to be inclusive (`== zf`) to avoid a discontinuity?

Reply:
I did fix that, but the issue was that I was using `z` (of sim) in half the profile and `myz` (an array at finer resolution at which to evaluate the profile function) in the other half. it's continuous now.